client/asset/zec/zec.go

```diff
@@ -2912,3 +3010,679 @@ func deserializeTx(b []byte) (*wire.MsgTx, error) {
+	txHistoryDB := w.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
+	err := txHistoryDB.MarkTxAsSubmitted(txHash.String())
+	if err != nil {
+		w.log.Errorf("failed to mark tx as submitted in tx history db: %v", err)
+	}
+
+	w.pendingTxsMtx.Lock()
+	wt, found := w.pendingTxs[*txHash]
+	w.pendingTxsMtx.Unlock()
+
+	if !found {
+		w.log.Errorf("Transaction %s not found in pending txs", txHash)
+		return
+	}
+
+	wt.Submitted = true
+
+	w.emit.TransactionNote(wt.WalletTransaction, true)
+}
+
+func (w *zecWallet) removeTxFromHistory(txHash *chainhash.Hash) {
+	txHistoryDB := w.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
+	w.pendingTxsMtx.Lock()
+	delete(w.pendingTxs, *txHash)
+	w.pendingTxsMtx.Unlock()
+
+	err := txHistoryDB.RemoveTx(txHash.String())
+	if err != nil {
+		w.log.Errorf("failed to remove tx from tx history db: %v", err)
+	}
+}
+
+func (w *zecWallet) addTxToHistory(wt *asset.WalletTransaction, txHash *chainhash.Hash, submitted bool, skipNotes ...bool) {
+	txHistoryDB := w.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
+	ewt := &btc.ExtendedWalletTx{
+		WalletTransaction: wt,
+		Submitted:         submitted,
+	}
+
+	if wt.BlockNumber == 0 {
+		w.pendingTxsMtx.Lock()
+		w.pendingTxs[*txHash] = ewt
+		w.pendingTxsMtx.Unlock()
+	}
+
+	err := txHistoryDB.StoreTx(ewt)
+	if err != nil {
+		w.log.Errorf("failed to store tx in tx history db: %v", err)
+	}
+
+	skipNote := len(skipNotes) > 0 && skipNotes[0]
+	if submitted && !skipNote {
+		w.emit.TransactionNote(wt, true)
+	}
+}
+
+func (w *zecWallet) txHistoryDBPath(walletID string) string {
+	return filepath.Join(w.walletDir, fmt.Sprintf("txhistorydb-%s", walletID))
+}
+
+// findExistingAddressBasedTxHistoryDB finds the path of a tx history db that
+// was created using an address controlled by the wallet. This should only be
+// used for RPC wallets, as SPV wallets are able to get the first address
+// generated by the wallet.
+func (w *zecWallet) findExistingAddressBasedTxHistoryDB() (string, error) {
+	dir, err := os.Open(w.walletDir)
+	if err != nil {
+		return "", fmt.Errorf("error opening wallet directory: %w", err)
+	}
+	defer dir.Close()
+
+	entries, err := dir.Readdir(0)
+	if err != nil {
+		return "", fmt.Errorf("error reading wallet directory: %w", err)
+	}
+
+	pattern := regexp.MustCompile(`^txhistorydb-(.+)$`)
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		match := pattern.FindStringSubmatch(entry.Name())
+		if match == nil {
+			continue
+		}
+
+		address := match[1]
+		owns, err := w.OwnsDepositAddress(address)
+		if err != nil {
+			continue
+		}
+		if owns {
+			return filepath.Join(w.walletDir, entry.Name()), nil
+		}
+	}
+
+	return "", nil
+}
+
+func (w *zecWallet) startTxHistoryDB(ctx context.Context) (*sync.WaitGroup, error) {
+	addressPath, err := w.findExistingAddressBasedTxHistoryDB()
+	if err != nil {
+		return nil, err
+	}
+	dbPath := addressPath
+
+	if dbPath == "" {
+		depositAddr, err := w.transparentAddress()
```

Comment:
Can we use the `mnemonic_seedfp` parameter of the[ `getwalletinfo` result](https://zcash.github.io/rpc/getwalletinfo.html).